sencha-workspace/packages/slate-cbl/src/view/student/skill/OverviewWindowController.js

```diff
@@ -0,0 +1,101 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.skill.OverviewWindowController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-skill-overviewwindow',
+    requires: [
+//        'Slate.cbl.API',
+//
+//        'Ext.MessageBox',
+//        'Ext.window.Toast'
+        'Slate.cbl.model.Competency',
+        'Slate.cbl.model.Skill'
+    ],
+
+    config: {
+        id: 'slate-cbl-student-skill-overviewwindow', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+                beforeshow: 'onBeforeWindowShow',
+                demorowclick: 'onDemoRowClick',
+                destroy: 'onDestroy'
+            },
+            'combobox[reference=skillCombo]': {
+                change: 'onSkillChange'
+            }
+        }
+    },
+
+    // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+    applyId: function(id) {
+        return Ext.id(null, id);
+    },
+
+    onBeforeWindowShow: function(overviewWindow) {
+        var me = this,
+            competencyId = overviewWindow.getCompetency(),
+            student = overviewWindow.getStudent(),
+            skillsCombo = me.lookupReference('skillCombo'),
+            skillStore = skillsCombo.getStore(),
+            skillId = overviewWindow.getSkill();
+
+        if (skillId && competencyId) {
+
+            if(!skillStore.isLoaded()) {
+                Slate.cbl.API.getSkills(competencyId, function(response){
+                    skillStore.loadRawData(response.data);
+
+                    skillsCombo.setValue(skillStore.findRecord('ID', skillId));
+                    me.syncDemonstrationsTable();
+                });
+
+            } else {
+                me.syncDemonstrationsTable();
+            }
+        }
+
+        Ext.on('resize', function() {
```

Comment:
I'm wary though of what behavior this solution causes when the overviewWindow is bigger than the browser viewport, I feel like it would prevent you from scrolling over its surface. We probably need to do something smarter that checks the bounds so its more like the top edge is "glued" to the top of the browser window when you scroll up above it, and scrolling down drags the bottom edge. Or we'd need to put a maximum height on the overviewWindow and enable inner scrolling somewhere.